discord/ext/converters/regex.py

```diff
@@ -0,0 +1,16 @@
+import re
+from typing import Union
+
+from discord.ext import commands
+
+
+class MatchConverter(commands.Converter):
+    def __init__(self, *, pattern: str, exception: str = "No match found.") -> None:
```

Comment:
You have the right idea but this is silly.

```suggestion
    def __init__(self, pattern: str) -> None:
```

It isn't a terrible idea to allow passing RegEx flags here too.

Reply:
Surely passing all args and kwargs to match should be best here